File: spx-gui/src/apis/traffic-source.ts
import { client } from './common'

export type TrafficSourceData = {
  /** Unique identifier */
  id: string
  /** Creation timestamp */
  createdAt: string
  /** Last update timestamp */
  updatedAt: string
  /** Source platform */
  platform: string
  /** Number of times this traffic source has been accessed */
  accessCount: number
}

/**
 * Create a new traffic source recording for sharing analytics
 */
export async function createTrafficSource(platform: string): Promise<TrafficSourceData> {
  return client.post(`/analytics/traffic-source`, platform)
}

/**
 * Record an access to existing traffic source
 */
export async function recordTrafficAccess(trafficSourceId: string): Promise<void> {
  return client.post(`/analytics/traffic-access`, trafficSourceId) as Promise<void>
}

File: spx-backend/internal/model/traffic_source.go
package model

// TrafficSource is the model for traffic source tracking records.
type TrafficSource struct {
	Model

	// Platform is the source platform.
	Platform string `gorm:"column:platform;size:50;index"`

	// AccessCount is the number of times this traffic source has been accessed.
	AccessCount int64 `gorm:"column:access_count;default:0;index"`
}

// TableName implements [gorm.io/gorm/schema.Tabler].
func (TrafficSource) TableName() string {
	return "traffic_source"
}
